### PR TITLE
Add support for flipping borrowed OpenGL textures vertically

### LINKED ISSUE #2986: Add support for importing Y-flipped OpenGL Textures
At the moment the API for importing an OpenGL texture into a `slint::Image` assumes that the origin of the texture is at the top-left. This is consistent with the coordinate system used in Slint, Vulkan, and generally what most people in computer graphics expect. However this may be inconvenient sometimes and it would be useful to have API that supports populating texture content with the origin being at the bottom-left (OpenGL default), by having a flag that tells slint to draw the texture y-flipped.

### PR DESCRIPTION
**Pull request by tronical**
Closes #2986

### CODE REVIEW

File: internal/core/graphics/image.rs [hunk 1]
```diff
@@ -686,6 +687,64 @@ impl Image {
...
+    /// The bottom-left of the texture is the top-left of the texture draw on the screen,
+    /// flipping it vertically.
+    BottomLeft,
+}
+
+/// Factory to create [`slint::Image`](crate::graphics::Image) from an existing OpenGL texture.
+///
+/// Methods can be chained on it in order to configure it.
+///
+///  * `mirror_vertically`: The texture is mirrored vertically when drawing. (default: false)
```
Comment (ogoffart) [inline internal/core/graphics/image.rs:706]: outdated comment

File: internal/core/graphics/image.rs [hunk 2]
```diff
@@ -686,6 +687,64 @@ impl Image {
     }
 }
 
+/// This enum describes the origin to use when rendering a borrowed OpenGL texture.
```
[the remark this answers is not in the document]
Comment (tronical) [reply internal/core/graphics/image.rs:690]: Good idea.

File: internal/core/graphics/image.rs (hunk 1 above)
Comment (tronical) [reply internal/core/graphics/image.rs:706]: Oops.

File: internal/core/graphics/image.rs [hunk 3]
```diff
@@ -686,6 +687,65 @@ impl Image {
...
+/// Use this with [BorrowedOpenGLTextureBuilder::origin].
```
Comment (ogoffart) [inline internal/core/graphics/image.rs:691]:
```suggestion
/// Use this with [`BorrowedOpenGLTextureBuilder::origin`].
```

**Review by ogoffart — APPROVED** [review]

File: internal/core/graphics/image.rs [hunk 4]
```diff
@@ -686,6 +687,65 @@ impl Image {
...
+    /// flipping it vertically.
+    BottomLeft,
+}
+
+/// Factory to create [`slint::Image`](crate::graphics::Image) from an existing OpenGL texture.
+///
+/// Methods can be chained on it in order to configure it.
+///
+///  * `origin`: Change the texture's origin when rendering (defaullt: TopLeft).
+///
```
Comment (ogoffart) [inline internal/core/graphics/image.rs:708]: Could there be a sample example of usage?

**Review by ogoffart — APPROVED** [review]

File: internal/core/graphics/image.rs (hunk 4 above)
Comment (tronical) [reply internal/core/graphics/image.rs:708]: I changed the texture import example to use the builder (that's how I tested it :), but for the flipping itself: I'd say it's a bit of a legacy feature. You integrate some third-party rendered content, find out that they use the original GL coordinate system, flip this switch, and that's it. Normally changing the origin should not be needed.

Comment (tronical) [reply internal/core/graphics/image.rs:708]: Ahh, yes, I can do that :)

**Pull request merged**